controller.py

```diff
@@ -0,0 +1,279 @@
+
+    def connect_to_shotgrid(self) -> None:
+        """Starts the ShotGrid model connection on a seperate thread."""
+        self.view.start_widget.hide()
+        self.view.layout.addWidget(self.view.get_loading_widget())
+
+        self.shotgrid_connection_thread = ShotGridConnectionThread(self.model)
+        self.shotgrid_connection_thread.connection_response_received.connect(
+            self.connection_response_received
+        )
+        self.shotgrid_connection_thread.start()
+
+    def connection_response_received(
+        self, connection_information: tuple(bool, str)
+    ) -> None:
+        """Runs when the ShotGrid connection thread is finished.
+        Shows error if needed, else moves on to finding the username.
+
+        Args:
+            connection_information: ShotGrid connection information
+        """
+        connected, error = connection_information
+
+        if connected:
+            self.find_username()
+        else:
+            self.view.loading_widget.hide()
+            self.view.layout.addWidget(self.view.get_error_widget())
+            self.view.error_text.setText(
+                f"Error: {error}. Please contact a pipeliner if problem persist."
+            )
+
+    def find_username(self) -> None:
+        """Checks the username with the model. If we can't find a ShotGrid username,
+        we show the username selection sceen to the user.
+        """
+        self.view.loading_widget.hide()
+
+        shotgrid_user = self.model.get_shotgrid_user_from_computer_username()
+
+        if shotgrid_user:
+            self.model.set_user_information(shotgrid_user)
+            self.view.layout.addWidget(
+                self.view.get_main_widget(
+                    shotgrid_user.get("name"), self.model.usernames
+                )
+            )
+        else:
+            self.view.layout.addWidget(
+                self.view.get_username_widget(self.model.usernames)
+            )
+
+    def validate_username(self) -> None:
+        """Checks if user submitted username is in ShotGrid. Moves on to next
+        step if username is correct."""
+        shotgrid_user = self.model.get_shotgrid_user(
+            self.view.username_lineedit.text()
+        )
+
+        if not shotgrid_user:
+            self.view.username_validation_text.setStyleSheet(
+                "color: '#FF3E3E'; font-size: 12px;"
+            )
+            self.view.username_validation_text.setText(
+                "Could not find user in ShotGrid database."
+            )
+            self.view.username_validation_text.show()
+
+        else:
+            self.view.username_widget.hide()
+            self.view.layout.addWidget(
+                self.view.get_main_widget(
+                    shotgrid_user.get("name"), self.model.usernames
+                )
+            )
+            self.model.set_user_information(shotgrid_user)
+
+    def validate_project_name(self, project_name: str) -> None:
```

Comment:
Simple things setting stylesheet according to text that has been set can be moved to the view itself